constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/TransitionParser.java

```diff
@@ -321,6 +330,71 @@ private static void parseKeyAttribute(CLObject keyAttribute,
                 }
             }
         }
+        // Support for custom attributes in KeyAttributes
+        CLElement customElement = keyAttribute.getOrNull("custom");
+        if (customElement != null && customElement instanceof CLObject) {
+            CLObject customObj = ((CLObject) customElement);
+            int n = customObj.size();
+            customVars =  new CustomVariable[frames.size()][n];
+            for (int i = 0; i < n; i++) {
+                CLKey key = (CLKey) customObj.get(i);
+                String customName = key.content();
+                Utils.log(" >>>>customObj  "+customObj.content());
+                Utils.log(" >>>>key "+key.content());
+                Utils.log(" >>>>key.value "+key.getValue());
+                Utils.log(" >>>> "+customName);
+
+                if (key.getValue() instanceof CLArray) {
+                    Utils.log(" >>>> "+ customName+" CLArray ");
+                    CLArray arrayValues = (CLArray) key.getValue() ;
+                    int vSize = arrayValues.size();
+                    if (vSize == bundles.length && vSize > 0) {
+                        if (arrayValues.get(0) instanceof CLNumber) {
```

Comment:
A brief comment would be helpful for understanding what is the if condition and else condition about.

Reply:
done